src/Agent/NewRelic/Agent/Core/Configuration/DefaultConfiguration.cs

```diff
@@ -183,17 +183,26 @@ private int TryGetAppSettingAsIntWithDefault(string key, int defaultValue)
 
         public object AgentRunId { get { return _serverConfiguration.AgentRunId; } }
 
+        private static bool? _agentEnabledAppSettingParsed;
```

Comment:
Does this field need to be `volatile` since it is only read outside of the lock?

Reply:
`bool?` can't be `volatile`. No idea why...

Reply:
That surprised me. Should we be checking the value again inside the lock just to be safe (double-checked lock pattern)?

Reply:
Yeah, I'm adding that now. Good catch!